booster/tools/rpc-client/RpcClient.hs

```diff
@@ -618,6 +582,97 @@ runTarball common (Just sock) tarFile keepGoing runOnly compareDetails = do
                             then pure Nothing
                             else pure . Just $ showResult (DifferentType expectedType actualType)
 
+{- | Given a bug report tarball, traverse the requests and their corresponding responses and for each pair, output:
+     * request type and response summary
+-}
+dryRunTarball ::
+    CommonOptions ->
+    FilePath ->
+    IO ()
+dryRunTarball common tarFile = do
+    -- unpack tar files, determining type from extension(s)
+    let unpackTar
+            | ".tar" == takeExtension tarFile = Tar.read
+            | ".tgz" == takeExtension tarFile = Tar.read . GZip.decompress
+            | ".tar.gz" `isSuffixOf` takeExtensions tarFile = Tar.read . GZip.decompress
+            | ".tar.bz2" `isSuffixOf` takeExtensions tarFile = Tar.read . BZ2.decompress
+            | otherwise = Tar.read
+
+    entries <- Tar.decodeLongNames . unpackTar <$> BS.readFile tarFile
+    withTempDir $ \tmpDir ->
+        withLogLevel common.logLevel $ do
+            -- unpack relevant tar files (rpc_* directories only)
+            logInfo_ $ unwords ["unpacking json files from tarball", tarFile, "into", tmpDir]
+            (jsonFiles, sequenceMap) <-
+                liftIO $ Tar.foldEntries (unpackIfRpc tmpDir) (pure mempty) (error . show) entries
+
+            -- we should not rely on the requests being returned in a sorted order and
+            -- should therefore sort them explicitly
+            let requests = mapMaybe (stripSuffix "_request.json") $ sortBy (compareSequence sequenceMap) jsonFiles
+
+            forM_ requests $ \basename -> do
+                request <- liftIO $ do
+                    let requestFile = tmpDir </> basename <> "_request.json"
+                    doesFileExist requestFile >>= \case
+                        True -> fmap Just . BS.readFile $ tmpDir </> basename <> "_request.json"
+                        False -> pure Nothing
+                expected <- liftIO $ do
+                    let responseFile = tmpDir </> basename <> "_response.json"
+                    doesFileExist responseFile >>= \case
+                        True -> fmap Just . BS.readFile $ tmpDir </> basename <> "_response.json"
+                        False -> pure Nothing
+                case (decodeKoreRpc <$> request, decodeKoreRpc <$> expected) of
+                    (Just (RpcRequest req), Just (RpcResponse resp)) -> do
+                        logInfo_ $
+                            renderDefault . Pretty.hsep . Pretty.punctuate Pretty.comma $
+                                [pretty basename, pretty req, pretty resp]
+                    (Just (RpcRequest req), _) -> do
+                        logInfo_ $
+                            renderDefault . Pretty.hsep . Pretty.punctuate Pretty.comma $
+                                [pretty basename, pretty req, "no response"]
+                    _ -> do
+                        logWarn_ "Expected an RPC request and response pair. Skipping..."
```

Comment:
Plenty of  duplicated code... this part is the actual difference (in comparison with `runAllRequests` above).
We can merge this but we will have to duplicate all code that handles future changes to the tarball format.